src/Microsoft.Framework.DependencyInjection.Abstractions/ServiceProviderExtensions.cs

```diff
@@ -82,7 +82,14 @@ public static IEnumerable<T> GetRequiredServices<T>([NotNull] this IServiceProvi
         public static IEnumerable<object> GetRequiredServices([NotNull] this IServiceProvider provider, [NotNull] Type serviceType)
         {
             var genericEnumerable = typeof(IEnumerable<>).MakeGenericType(new [] { serviceType });
-            return (IEnumerable<object>)provider.GetRequiredService(genericEnumerable);
+            var providers = (IEnumerable<object>)provider.GetRequiredService(genericEnumerable);
+            
+            if (!providers.Any())
```

Comment:
I'm probably fine with this as-is, but maybe log a bug to look at that in the future. Ultimately we'd have to run this in a profiler to see if this even matters.